From a67b3085d5b5c329071d726a5dc9a3ea0db05b2f Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megous@megous.com>
Date: Fri, 22 Oct 2021 18:11:26 +0200
Subject: [PATCH 463/467] media: i2c: imx258: Don't be too strict about clock
 rate

On Pinephone Pro, we are not able to set 19.2MHz precisely.
Allow some slack in clock rate.

Signed-off-by: Ondrej Jirman <megous@megous.com>
---
 drivers/media/i2c/imx258.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/media/i2c/imx258.c b/drivers/media/i2c/imx258.c
index be5adcccd..0ee93aa1f 100644
--- a/drivers/media/i2c/imx258.c
+++ b/drivers/media/i2c/imx258.c
@@ -79,7 +79,9 @@
 #define REG_CONFIG_FLIP_TEST_PATTERN	0x02
 
 /* Input clock frequency in Hz */
+#define IMX258_INPUT_CLOCK_FREQ_MIN	19000000
 #define IMX258_INPUT_CLOCK_FREQ		19200000
+#define IMX258_INPUT_CLOCK_FREQ_MAX	19400000
 
 struct imx258_reg {
 	u16 address;
@@ -1277,7 +1279,8 @@ static int imx258_probe(struct i2c_client *client)
 	} else {
 		val = clk_get_rate(imx258->clk);
 	}
-	if (val != IMX258_INPUT_CLOCK_FREQ) {
+	if (val < IMX258_INPUT_CLOCK_FREQ_MIN
+		|| val > IMX258_INPUT_CLOCK_FREQ_MAX) {
 		dev_err(&client->dev, "input clock frequency not supported\n");
 		return -EINVAL;
 	}
-- 
2.34.0

